src/v2i-hub/CARMAStreetsPlugin/src/J2735MapToJsonConverter.cpp

```diff
@@ -0,0 +1,267 @@
+
+                    ss.str("");
+                    ss << node->delta.choice.node_XY2.y;
+                    nodeJson["delta"]["node-xy"]["y"] = ss.str();
+                    nodeListJson.append(nodeJson);
+                }
+                else if (node->delta.present == NodeOffsetPointXY_PR::NodeOffsetPointXY_PR_node_XY3)
+                {
+                    ss.str("");
+                    ss << node->delta.choice.node_XY3.x;
+                    nodeJson["delta"]["node-xy"]["x"] = ss.str();
+
+                    ss.str("");
+                    ss << node->delta.choice.node_XY3.y;
+                    nodeJson["delta"]["node-xy"]["y"] = ss.str();
+                    nodeListJson.append(nodeJson);
+                }
+                else if (node->delta.present == NodeOffsetPointXY_PR::NodeOffsetPointXY_PR_node_XY4)
+                {
+                    ss.str("");
+                    ss << node->delta.choice.node_XY4.x;
+                    nodeJson["delta"]["node-xy"]["x"] = ss.str();
+
+                    ss.str("");
+                    ss << node->delta.choice.node_XY4.y;
+                    nodeJson["delta"]["node-xy"]["y"] = ss.str();
+                    nodeListJson.append(nodeJson);
+                }
+                else if (node->delta.present == NodeOffsetPointXY_PR::NodeOffsetPointXY_PR_node_XY5)
+                {
+                    ss.str("");
+                    ss << node->delta.choice.node_XY5.x;
+                    nodeJson["delta"]["node-xy"]["x"] = ss.str();
+
+                    ss.str("");
+                    ss << node->delta.choice.node_XY5.y;
+                    nodeJson["delta"]["node-xy"]["y"] = ss.str();
+                    nodeListJson.append(nodeJson);
+                }
+                else if (node->delta.present == NodeOffsetPointXY_PR::NodeOffsetPointXY_PR_node_XY6)
+                {
+                    ss.str("");
+                    ss << node->delta.choice.node_XY6.x;
+                    nodeJson["delta"]["node-xy"]["x"] = ss.str();
+
+                    ss.str("");
+                    ss << node->delta.choice.node_XY6.y;
+                    nodeJson["delta"]["node-xy"]["y"] = ss.str();
+                    nodeListJson.append(nodeJson);
+                }
+            }
+        }
+    }
+
+    void J2735MapToJsonConverter::convertConnectsToJSON(const GenericLane *lane, Json::Value &connectsJson) const
+    {
+        std::stringstream ss;
+        if (lane->connectsTo != nullptr)
+        {
+            for (size_t i = 0; i < lane->connectsTo->list.count; i++)
+            {
+                Json::Value connJson;
+                ss.str("");
+                auto connect = lane->connectsTo->list.array[i];
+                ss << connect->connectingLane.lane;
+                connJson["connecting_lane"]["lane"] = ss.str();
+
+                ss.str("");
+                connect = lane->connectsTo->list.array[i];
+                ss << *connect->signalGroup;
```

Comment:
I am not clear why you sometimes need to deference the Map object or Map Object elements and other times not. Are there specific types this is necessary for?

Reply:
I only dereference it when it is a pointer type. Not when it is not a pointer. This is from the J2735 MAP message definition.